src/common/users.ts

```diff
@@ -114,6 +125,94 @@ export interface ReadingDaysArgs {
   dateRange: DateRange;
 }
 
+interface RecentMentionsProps {
+  query?: string;
+  limit?: number;
+  excludeUsernames?: string[];
+}
+
+export const getRecentMentions = (
+  con: Connection,
+  userId: string,
+  { limit = 5, query, excludeUsernames }: RecentMentionsProps,
+): Promise<MentionUser[]> => {
+  let queryBuilder = con
+    .getRepository(CommentMention)
+    .createQueryBuilder('cm')
+    .select('DISTINCT u.name, u.username, u.image')
+    .innerJoin(Comment, 'c', 'cm."commentId" = c.id')
+    .innerJoin(DbUser, 'u', 'u.id = cm."mentionedUserId"')
+    .where('c."userId" = :userId', { userId })
+    .andWhere('cm."mentionedUserId" != :userId', { userId })
+    .limit(limit);
```

Comment:
I agree, that would make it much easier in terms of querying.

Reply:
That makes total sense, let me check if I can do the inner join even if I already finished writing the where clause since there can be a condition where we can to search for the user's query.